internal/terraform/lang/parser.go

```
@@ -131,26 +132,28 @@ func (bt *completableBlockType) Documentation() MarkupContent {
 	return bt.documentation
 }
 
-func (p *parser) ParseBlockFromHCL(block *hcl.Block) (ConfigBlock, error) {
-	if block == nil {
+func (p *parser) ParseBlockFromTokens(tokens hclsyntax.Tokens) (ConfigBlock, error) {
+	if len(tokens) == 0 {
 		return nil, EmptyConfigErr
 	}
 
+	// It is probably excessive to be parsing the whole block just for type
+	// but there is no avoiding it without refactoring the upstream HCL parser
+	// and it should not hurt the performance too much
+	block, _ := hclsyntax.ParseBlockFromTokens(tokens)
```

Same as above, could you expand the comment here about why ignoring the error?